clients/blocksdata/r2_client.go

```diff
@@ -54,22 +58,34 @@ func (c *blocksDataClient) GetBlocksData(bucket int64) (_ *protocol.BlocksData,
 	var item PresignedURLItem
 
 	err = backoff.Retry(func() error {
-		resp, err := httpclient.Default.Get(dispatcherUrl)
+		req, err := http.NewRequest(http.MethodGet, dispatcherUrl, nil)
 		if err != nil {
-			return err
+			return backoff.Permanent(err)
 		}
 
-		defer resp.Body.Close()
+		scannerJwt, err := security.CreateScannerJWT(c.key, nil)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
 
-		b, err := io.ReadAll(resp.Body)
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", scannerJwt))
+
+		resp, err := httpclient.Default.Get(dispatcherUrl)
```

Comment:
no one uses the request....

Reply:
(Please note the difference between `httpclient.Default` and `http.DefaultClient` - the former is from our core repo and has some defaults.)

Reply:
yes, the constructed request is not used and therefore the requests don't have the auth bearer header - client is making a generic GET call to dispatcher